src/runtime/hexagon_remote/remote.cpp

```diff
@@ -44,15 +95,239 @@ int halide_do_task(void *user_context, halide_task_t f, int idx,
                    uint8_t *closure) {
     return f(user_context, idx, closure);
 }
+void lock_work_queue() {
+    qurt_mutex_lock(&(work_queue.work_mutex));
+}
+void unlock_work_queue() {
+    qurt_mutex_unlock(&(work_queue.work_mutex));
+}
+void goto_work(work *owned_job) {
+    work myjob;
```

Comment:
Why does this need to make a local copy of the work? I think it should be fine to just let orig_job (or job_to_do, see my comment below) point to the work. If I'm wrong about this, a comment on why the work needs to be copied locally would be good.